apps/client/src/features/app-settings/panel/client-control-panel/ClientList.tsx

```diff
@@ -0,0 +1,205 @@
+
+  const openRename = (id: string) => {
+    setTargetId(id);
+    onOpenRename();
+  };
+
+  const rename = (name: string) => {
+    setRename({ target: targetId, name });
+    onCloseRename();
+  };
+
+  const openRedirect = (id: string) => {
+    setTargetId(id);
+    onOpenRedirect();
+  };
+
+  const redirect = (path: string) => {
+    setRedirect({ target: targetId, path });
+    onCloseRedirect();
+  };
+
+  return (
+    <>
+      <RedirectModal
+        onClose={onCloseRedirect}
+        isOpen={isOpenRedirect}
+        clients={clients}
+        id={targetId}
+        onSubmit={redirect}
+      />
+      <RenameModal onClose={onCloseRename} isOpen={isOpenRename} clients={clients} id={targetId} onSubmit={rename} />
+      <Panel.Table>
+        <thead>
+          <tr>
+            <td className={style.fullWidth}>Client Name (Connection ID)</td>
+            <td />
+          </tr>
+        </thead>
+        <tbody>
+          {Object.entries(clients).map(([id, client]) => {
+            const { identify, redirect, name } = client;
+            const isCurrent = id === myId;
+            const isRedirecting = redirect != '';
+            return (
+              <tr key={id} className={isCurrent ? style.current : undefined}>
+                <td className={style.badgeList}>
+                  <Badge variant='outline' size='sx'>
+                    {id}
+                  </Badge>
+                  <Badge hidden={!isCurrent} variant='outline' colorScheme='yellow' size='sx'>
+                    self
+                  </Badge>
+                  {name}
+                </td>
+                <td className={style.actionButtons}>
+                  <Button
+                    size='xs'
+                    className={`${identify ? style.blink : ''}`}
+                    variant={identify ? 'ontime-filled' : 'ontime-subtle'}
+                    onClick={() => {
+                      setIdentify({ target: id, state: !identify });
+                    }}
+                    isActive
+                  >
+                    Identify
+                  </Button>
+                  <Button size='xs' variant='ontime-subtle' onClick={() => openRename(id)}>
+                    Rename
+                  </Button>
+
+                  <ButtonGroup size='xs' isAttached variant='ontime-subtle'>
+                    <Button isLoading={isRedirecting} onClick={() => openRedirect(id)}>
+                      Redirect
+                    </Button>
+                    {isRedirecting && (
+                      <IconButton
+                        aria-label='Cancel the redirect'
+                        icon={<IoClose />}
+                        onClick={() => setRedirect({ target: id, path: '' })}
+                      />
+                    )}
+                  </ButtonGroup>
+                </td>
+              </tr>
+            );
+          })}
+        </tbody>
+      </Panel.Table>
+    </>
+  );
+}
+
+function RedirectModal(props: {
```

Comment:
suggestion: lets extract this into another file. I think it may be easier for us